app/javascript/controllers/evaluation_criteria_controller.js

```diff
@@ -0,0 +1,221 @@
+// app/javascript/controllers/evaluation_criteria_controller.js
+import { Controller } from "@hotwired/stimulus";
+
+export default class extends Controller {
+  static targets = [
+    "criteriaList",
+    "template",
+    "addButton",
+    "criteriaRow",
+    "scoringRadio",
+    "binaryOptions",
+    "ratingOptions",
+    "scaleOptions",
+    "hiddenOptionRangeStart",
+    "hiddenOptionRangeEnd",
+    "selectOptionRangeStart",
+    "selectOptionRangeEnd",
+    "criteriaLabelRow",
+  ];
+
+  connect() {
+    this.counter = this.criteriaRowTargets.length;
+  }
+
+  addCriteria() {
+    this.counter++;
+    const newCriteria = this.templateTarget.cloneNode(true);
+
+    this.replacePlaceholders(newCriteria);
+    this.enableInputs(newCriteria);
+
+    this.criteriaListTarget.appendChild(newCriteria);
+
+    this.updateCriteriaTitles();
+  }
+
+  removeCriteria(event) {
```

Comment:
It doesn't really matter since it's mostly there for making sure the submitted nested fields have a unique ID for each criteria. It could cause conflicts with that uniqueness when decrementing. 

Removed fields that were presenting when updating remain in the form with a hidden `id` and `_destroy` field to make sure they get removed from the database.